src/neurostatslib/glm.py

```diff
@@ -1,431 +1,598 @@
+
+        $$
+        \begin{aligned}
+            R^2_{\text{pseudo}} &= \frac{D_{\text{null}} - D_{\text{model}}}{D_{\text{null}}} \\\
+            &= \frac{\log \text{LL}(\hat{\lambda}| y) - \log \text{LL}(\bar{\lambda}| y)}{\log \text{LL}(y| y)
+            - \log \text{LL}(\bar{\lambda}| y)},
+        \end{aligned}
+        $$
 
-    def predict(self, spike_data: NDArray) -> jnp.ndarray:
-        """Predict firing rates based on fit parameters, for checking against existing data.
+        where LL is the log-likelihood, $D_{\text{null}}$ is the deviance for a null model, $D_{\text{model}}$ is
+        the deviance for the current model, $y_{tn}$ and $\hat{\lambda}_{tn}$ are the observed activity and the model
+        predicted rate for neuron $n$ at time-point $t$, and $\bar{\lambda}$ is the mean firing rate. See [1].
 
         Parameters
         ----------
-        spike_data : (n_neurons, n_timebins)
-            Spike counts arranged in a matrix. n_neurons must be the same as
-            during the fitting of this GLM instance.
+        X :
+            The exogenous variables. Shape (n_time_bins, n_neurons, n_features)
+        y :
+            Neural activity arranged in a matrix. n_neurons must be the same as
+            during the fitting of this GLM instance. Shape (n_time_bins, n_neurons).
+        score_type :
+            Type of scoring: either log-likelihood or pseudo-r2.
 
         Returns
         -------
-        predicted_firing_rates : (n_neurons, n_timebins - window_size + 1)
-            The predicted firing rates.
+        score :
+            The log-likelihood or the pseudo-$R^2$ of the current model.
 
         Raises
         ------
         NotFittedError
             If ``fit`` has not been called first with this instance.
         ValueError
             If attempting to simulate a different number of neurons than were
-            present during fitting (i.e., if ``init_spikes.shape[0] !=
-            self.baseline_log_fr_.shape[0]``).
+            present during fitting (i.e., if ``init_y.shape[0] !=
+            self.baseline_link_fr_.shape[0]``).
 
-        See Also
-        --------
-        score
-            Score predicted firing rates against target spike counts.
-        simulate
-            Simulate spikes using GLM as a recurrent network, for extrapolating into the future.
+        Notes
+        -----
+        The log-likelihood is not on a standard scale, its value is influenced by many factors,
+        among which the number of model parameters. The log-likelihood can assume both positive
+        and negative values.
+
+        The Pseudo-$R^2$ is not equivalent to the $R^2$ value in linear regression. While both provide a measure
+        of model fit, and assume values in the [0,1] range, the methods and interpretations can differ.
+        The Pseudo-$R^2$ is particularly useful for generalized linear models where a traditional $R^2$ doesn't apply.
```

Comment:
I added a long discussion about why residuals and r^2 cannot be interpreted in the same way for glms. I tried to provide examples that clarifies the issue; let me know if it is good enough

Reply:
agree, I'll leave it there for now because we don't have a theory section or a note on evaluation metrics.
we may think of a notebook in which we compare evaluation metrics on different syntehtic and real data fits. this would be very useful but it is almost a book chapter. I am sure we could do it having Alex reviewing it/guiding us